src/corelibs/U2Core/src/datatype/msa/MultipleSequenceAlignmentRow.cpp

```diff
@@ -557,4 +566,54 @@ MultipleAlignmentData *MultipleSequenceAlignmentRowData::getMultipleAlignmentDat
     return alignment;
 }
 
+const QVector<U2MsaGap> &MultipleSequenceAlignmentRowData::getGaps() const {
+    return gaps;
+}
+
+const DNASequence &MultipleSequenceAlignmentRowData::getSequence() const {
+    return sequence;
+}
+
+qint64 MultipleSequenceAlignmentRowData::getRowLengthWithoutTrailing() const {
+    return MsaRowUtils::getRowLength(sequence.seq, gaps);
+}
+
+int MultipleSequenceAlignmentRowData::getCoreEnd() const {
+    return getRowLengthWithoutTrailing();
+}
+
+bool MultipleSequenceAlignmentRowData::simplify() {
+    CHECK(!gaps.isEmpty(), false);
+    invalidateGappedCache();
+    gaps.clear();
+    return true;
+}
+
+int MultipleSequenceAlignmentRowData::getGapsLength() const {
+    return MsaRowUtils::getGapsLength(gaps);
+}
+
+void MultipleSequenceAlignmentRowData::invalidateGappedCache() const {
+    gappedCacheOffset = 0;
+    gappedSequenceCache.clear();
+}
+
+/**
+ * Size of the gapped cache: 200 symbols.
+ * Optimized for chars per screen and makes getChar() ~200x faster thanMsaRowUtils::charAt for sequential access.
+ */
+static constexpr int GAPPED_CACHE_SIZE = 200;
+
+char MultipleSequenceAlignmentRowData::getCharFromCache(int gappedPosition) const {
+    if (gappedPosition >= gappedCacheOffset && gappedPosition < gappedCacheOffset + gappedSequenceCache.size()) {
```

Comment:
Don't like this way of inequalities ordering. You need to look for some time on it to understand, how parts of the inequality relates to each other. Much better is to use the only `<` and `<=` signs and write variables from smallest to largest:
`gappedCacheOffset <= gappedPosition && gappedPosition < gappedCacheOffset + gappedSequenceCache.size()`

Here you may see easily, that `gappedPosition` should be between `gappedCacheOffset` and `gappedCacheOffset + gappedSequenceCache.size()`

Reply:
I do not agree. The checked subject (gappedPosition) should be always on the same side.

Reply:
Let us both do in a way we like

Reply:
Ok, I never have problems like this in reviews: we do not have a strict code style about it.